### PR TITLE
initial implementation

### PR DESCRIPTION
**Pull request by robertkowalski**
implements a publisher that watches a directory for dazaar cards,
puts them into a hyperbee db and publishes them on the dazaar
network.

after startup the dazaar card / id of the publisher is printed to
the console

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    for (const fname of allFiles) {
+      if (path.extname(fname) !== '.json') {
+        continue
+      }
+
+      const f = path.join(directory, fname)
+      const c = await readFile(f, 'utf8')
+      const cj = JSON.parse(c)
+
+      await batch.put(fname, cj)
```
Comment (robertkowalski) [inline index.js:85]: @mafintosh is there something i can use to put without creating a new sequence?

basically sugar for:
```js
const entry = await db.get('foo')
if (entry) return
entry.put('foo', { bar: 'baz' })
```

core after 4 inserts currently:
```

hyperbee
foo.json
{"foo":"bar"}


foo.json
{"foo":"bar"}
foo.json
{"foo":"bar"}
foo.json
{"foo":"bar"}
```

Comment (robertkowalski) [reply index.js:85]: feature added in hyperbee 1.0.1! 👍

File: index.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+    const allFiles = await readdir(directory)
+    for (const fname of allFiles) {
+      if (path.extname(fname) !== '.json') {
+        continue
+      }
+
+      const f = path.join(directory, fname)
+      const c = await readFile(f, 'utf8')
+      const cj = JSON.parse(c)
```
Comment (mafintosh) [inline index.js]: you might wanna try-catch this in case the file is partially written and skip it if it doesn't parse

File: index.js [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+      let c
+      try {
+        c = await readFile(f, 'utf8')
+      } catch (e) {
+        if (e.code === 'ENOENT') {
+          return
+        }
+      }
+
+      const cj = JSON.parse(c)
```
Comment (mafintosh) [inline index.js]: same as above

**Comment by mafintosh** [comment]
Looks good other than my comment :+1:

**Pull request merged**